multifile_reader/multifile_reader.py

```diff
@@ -104,11 +149,16 @@ def read(self, size=None):
 
     def _read(self, size):
         unread = (
-            self._map_sizes[self._file.name]
-            - self._read_files_offset[self._file.name]
+            self._map_sizes[self.filename]
+            - self._read_files_offset[self.filename]
         )
         length = min(size, unread)
-        self._file.seek(self._read_files_offset[self._file.name])
-        buf = self._file.read(length)
-        self._read_files_offset[self._file.name] += length
+
+        if self._url:
+            buf = self._file.read()
```

Comment:
I think this doesnt read the file in batches, which is different from how the files are read from local file system. 
Can you add reading in batches based on `size` provided? 
This will provide a unified handling of files. Also it will help with other tools like boto3, which reads the file in chunks and uploads them separately, sometimes even in parallel and out of order.

Reply:
Yeah, I know. I'll see how could something like that be done. Maybe have a separate method for reading local files and online files, because mixing these two concepts is not that transparent in the code.